src/katgpucbf/xbgpu/bsend.py

```diff
@@ -194,30 +197,41 @@ def _inc_counters(
             format of (n_channels_per_substream, samples_per_spectra, COMPLEX).
         data_dtype
             The `np.dtype` of the beam data transmitted.
+        enabled
+            Boolean flag array indicating which streams are enabled for transmission.
         output_names
-            List of beam stream names that are enabled for transmission for
-            this `future`.
+            List of beam stream names
         saturated
             Saturation count for the chunk for each stream in `output_names`.
+        sensors
+            Server's katcp sensors
+        sensor_timestamp
+            Timestamp (UNIX time) to use for sensor update
         future
             Future returned by the spead2 stream's `async_send_heaps`.
 
         .. todo::
 
             (NGC-1172) Handle missing data transmission for Prometheus counters.
         """
-        if not future.cancelled() and future.exception() is None:
-            # int casts are because np.prod returns np.int64 which is
-            # incompatible with the type annotations for Prometheus.
-            byte_count = int(np.prod(data_shape)) * data_dtype.itemsize * n_frames_sent
-            sample_count = int(np.prod(data_shape[:-1])) * n_frames_sent
-            for i, output_name in enumerate(output_names):
+        if future.cancelled() or future.exception() is not None:
+            # Don't update output counters if we didn't successfully transmit the data.
+            n_frames_sent = 0
+        # int casts are because np.prod returns np.int64 which is
+        # incompatible with the type annotations for Prometheus.
+        # Multiply across dimensions to get total bytes
+        byte_count = int(np.prod(data_shape)) * data_dtype.itemsize * n_frames_sent
+        # Multiply across the first two dimensions to get complex sample count
+        sample_count = int(np.prod(data_shape[:-1])) * n_frames_sent
+        for i, output_name in enumerate(output_names):
+            clipped = int(saturated[i])  # Convert from numpy type to plain Python type
```

Comment:
Really bikeshedding now. Do you think we can consolidate this conversing/casting comment with the old ones above? Line 220+ kinda discuss the same thing.

Reply:
I'll just delete the comment here since it's redundant.